clients/python/client/osparc/_http_client.py

```diff
@@ -41,3 +44,44 @@ async def __aexit__(self, exc_type, exc_value, traceback) -> None:
                     raise err from exc_value
             await self._client.aclose()
             raise exc_value
+
+    async def _request(
+        self, method: Callable[[Any], Awaitable[httpx.Response]], *args, **kwargs
+    ) -> httpx.Response:
+        @tenacity.retry(
```

Comment:
What about the case in which you have a 503 with retry-after header in place? perhaps you should add a tenacity hook for that case?

Reply:
Yep, I added this case along with some tests as well